circleci: failed tests (12542): TestClientRetryNonTxn

The following test appears to have failed:

[#12542](https://circleci.com/gh/cockroachdb/cockroach/12542):

```
 * 16:21:24.196654 meta descriptor lookup
 * 16:21:24.196663 sending RPC
 * 16:21:24.196707 sending to 127.0.0.1:45550
 * 16:21:24.201384 reply error: *roachpb.WriteIntentError
 * 16:21:24.201426 error: conflicting intents on ["key-3"]: resolved? false
panic: sync: negative WaitGroup counter

goroutine 322 [running]:
sync.(*WaitGroup).Add(0xc8200d46c0, 0xffffffffffffffff)
	/usr/local/go/src/sync/waitgroup.go:70 +0xd0
sync.(*WaitGroup).Done(0xc8200d46c0)
	/usr/local/go/src/sync/waitgroup.go:95 +0x2a
github.com/cockroachdb/cockroach/client_test.(*notifyingSender).Send(0xc820208100, 0x7f2744b44dd0, 0xc820015810, 0x0, 0x0, 0x0, 0x0, 0x0, 0xbff0000000000000, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/client/client_test.go:71 +0x1a1
github.com/cockroachdb/cockroach/client.(*DB).send(0xc820207d10, 0xc8200d4ae0, 0x1, 0x1, 0x8, 0x63aadf)
	/go/src/github.com/cockroachdb/cockroach/client/db.go:482 +0x16b
github.com/cockroachdb/cockroach/client.(*DB).(github.com/cockroachdb/cockroach/client.send)-fm(0xc8200d4ae0, 0x1, 0x1, 0x0, 0xc8200d4ae0)
	/go/src/github.com/cockroachdb/cockroach/client/db.go:450 +0x3e
--
github.com/cockroachdb/cockroach/client_test.TestClientRetryNonTxn.func1.1(0xc82023e040, 0xc82025d360, 0xc820207d10, 0xc820186560, 0x5, 0x8, 0xc8202426c0, 0xc82015e000)
	/go/src/github.com/cockroachdb/cockroach/client/client_test.go:183 +0x491
created by github.com/cockroachdb/cockroach/client_test.TestClientRetryNonTxn.func1
	/go/src/github.com/cockroachdb/cockroach/client/client_test.go:193 +0x2a3

goroutine 1 [chan receive]:
testing.RunTests(0x1704aa8, 0x1d939a0, 0x1b, 0x1b, 0x1020e6501)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc82042ff08, 0x13bf1c0)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc82042ff08)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x21
github.com/cockroachdb/cockroach/client_test.TestMain(0xc82042ff08)
	/go/src/github.com/cockroachdb/cockroach/client/main_test.go:36 +0x21
main.main()
	github.com/cockroachdb/cockroach/client/_test/_testmain.go:126 +0x113

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

--- FAIL: TestClientRetryNonTxn (0.35s)
goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x20e0d20)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0xbc

goroutine 15 [runnable]:
syscall.Syscall(0x1, 0x1, 0xc820111b00, 0x28, 0x28, 0x28, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.write(0x1, 0xc820111b00, 0x28, 0x240, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:1064 +0x5f
syscall.Write(0x1, 0xc820111b00, 0x28, 0x240, 0x16ff0e9, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:176 +0x4d
os.(*File).write(0xc820028010, 0xc820111b00, 0x28, 0x240, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:232 +0xaa
os.(*File).Write(0xc820028010, 0xc820111b00, 0x28, 0x240, 0x4, 0x0, 0x0)
	/usr/local/go/src/os/file.go:139 +0x8a
--
testing.tRunner(0xc82015e000, 0x1d93b20)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 16 [select]:
github.com/cockroachdb/cockroach/rpc.(*RemoteClockMonitor).MonitorRemoteOffsets(0xc82025cd00, 0xc820224380)
	/go/src/github.com/cockroachdb/cockroach/rpc/clock_offset.go:156 +0x742
github.com/cockroachdb/cockroach/server.NewServer.func1()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:130 +0x31
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82025cd20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 34 [select]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:213 +0x1b6
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82025ce60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 35 [select]:
github.com/cockroachdb/cockroach/util.NewFeed.func1()
	/go/src/github.com/cockroachdb/cockroach/util/feed.go:61 +0x157
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82014e7c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 36 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc82023d2c0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:225 +0xca5
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:212 +0x20
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82014e8b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 37 [select]:
github.com/cockroachdb/cockroach/sql.(*LeaseManager).RefreshLeases.func1()
	/go/src/github.com/cockroachdb/cockroach/sql/lease.go:601 +0xdd0
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820257950)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 38 [select]:
github.com/cockroachdb/cockroach/sql.NewExecutor.func1()
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:175 +0x134
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820257ad0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 39 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8202594f0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 40 [IO wait]:
net.runtime_pollWait(0x7f27493d9070, 0x72, 0xc8200141a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820250ae0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820250ae0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820250a80, 0x0, 0x7f2744b479b0, 0xc82042c580)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820120268, 0xc800000000, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820257d70)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 41 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:72 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82025d100)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 42 [chan receive]:
github.com/cockroachdb/cockroach/gossip.(*server).start.func2()
	/go/src/github.com/cockroachdb/cockroach/gossip/server.go:262 +0x57
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82025d140)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 43 [select]:
github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
	/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:770 +0x207
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82025d160)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 44 [select]:
github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
	/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:806 +0x33a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc82025d180)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 83 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startPublishStatuses.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:511 +0x23e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7340)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 67 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1642 +0x85f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820015e50)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 68 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:582 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d6700)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 69 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:642 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d6720)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 70 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:661 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d6740)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 84 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:479 +0x3f1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7360)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 76 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:352 +0x2ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d71e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 77 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:352 +0x2ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7200)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 78 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:352 +0x2ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7220)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 79 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:352 +0x2ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7240)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 80 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:352 +0x2ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7260)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 81 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:352 +0x2ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7280)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 82 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820311bc0, 0xece4c0903, 0x33a67187, 0x20e05a0, 0xc8202572c0, 0xc820224380, 0xc82005c1a0, 0xc820027698)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc82005c1a0, 0xc8203a8300)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:214 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc8203a8450, 0xc8204123c0)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:223 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:216 +0x1e1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d72a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 85 [select]:
github.com/cockroachdb/cockroach/ts.(*poller).start.func1()
	/go/src/github.com/cockroachdb/cockroach/ts/db.go:76 +0x141
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820407700)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 86 [select]:
github.com/cockroachdb/cockroach/ts.(*poller).start.func1()
	/go/src/github.com/cockroachdb/cockroach/ts/db.go:76 +0x141
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820407710)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 87 [select]:
github.com/cockroachdb/cockroach/server.(*Server).startWriteSummaries.func1()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:302 +0x169
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820407720)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 88 [select]:
github.com/cockroachdb/cockroach/sql.(*SchemaChangeManager).Start.func1()
	/go/src/github.com/cockroachdb/cockroach/sql/schema_changer.go:487 +0x16fc
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7560)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 89 [IO wait]:
net.runtime_pollWait(0x7f27493d8fb0, 0x72, 0xc8200141a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8203c3560, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8203c3560, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8203c3500, 0x0, 0x7f2744b479b0, 0xc82042c480)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820028598, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8202d7760)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 90 [chan receive]:
github.com/cockroachdb/cockroach/sql/pgwire.(*Server).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/sql/pgwire/server.go:91 +0x4a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820407af0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 91 [select]:
github.com/cockroachdb/cockroach/rpc.(*Client).runHeartbeat(0xc82009afc0, 0x3b9aca00, 0x6fc23ac00, 0x4000000000000000, 0x0, 0x0, 0xc82011a4e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/client.go:347 +0x77b
github.com/cockroachdb/cockroach/rpc.NewClient.func1()
	/go/src/github.com/cockroachdb/cockroach/rpc/client.go:167 +0x36
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc820412a00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 58 [select]:
github.com/cockroachdb/cockroach/storage.(*idAllocator).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/id_alloc.go:129 +0x911
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820224380, 0xc8201ca620)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62

goroutine 52 [IO wait]:
net.runtime_pollWait(0x7f27493d8ef0, 0x72, 0xc8200141a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82043a1b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82043a1b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82043a150, 0xc82008fc00, 0x400, 0x400, 0x0, 0x7f27493d0118, 0xc8200141a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820028700, 0xc82008fc00, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc8200d6840)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 97 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc820253440, 0x7f2744b4c958, 0xc820420480, 0xc82011aea0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:336 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc820253440, 0x7f2744b4c958, 0xc820420480, 0xc82011aea0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:243 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:244 +0x89c

goroutine 98 [IO wait]:
net.runtime_pollWait(0x7f27493d8e30, 0x72, 0xc8200141a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82043a220, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82043a220, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82043a1c0, 0xc82008f400, 0x400, 0x400, 0x0, 0x7f27493d0118, 0xc8200141a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820028708, 0xc82008f400, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc820253440, 0x7f2744b48268, 0xc82041e840, 0x7f2744b4c958, 0xc820420480, 0xc820390590, 0xc82011aea0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:246 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:249 +0x90c

goroutine 321 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).heartbeatLoop(0xc82023d2c0, 0xc8200d4440, 0x10)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:579 +0x602
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).updateState.func1.1()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:758 +0x3f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc820224380, 0x1aa097b, 0x16, 0x2f7, 0xc820208700)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:141 +0xc9
FAIL	github.com/cockroachdb/cockroach/client	0.427s
?   	github.com/cockroachdb/cockroach/cmd/monitor	[no test files]
?   	github.com/cockroachdb/cockroach/cmd/protoc-gen-gogoroach	[no test files]
=== RUN   TestObjectIDForKey
--- PASS: TestObjectIDForKey (0.01s)
=== RUN   TestGet
--- PASS: TestGet (0.00s)
=== RUN   TestGetLargestID
--- PASS: TestGetLargestID (0.00s)
=== RUN   TestComputeSplits
E0209 16:21:28.796313 200 config/config.go:352  unable to determine largest reserved object ID from system config: descriptor table not found in system config of 0 values

```
Please assign, take a look and update the issue accordingly.

--- a/client/client_test.go
+++ b/client/client_test.go
@@ -48,18 +48,12 @@ var testUser = server.TestUser
 // (on call to reset()) for clients which need to wait on a command
 // being sent.
 type notifyingSender struct {
-	waiter  *sync.WaitGroup
+	notify  chan struct{}
 	wrapped client.Sender
 }
 
-func (ss *notifyingSender) reset(waiter *sync.WaitGroup) {
-	waiter.Add(1)
-	ss.waiter = waiter
-}
-
-func (ss *notifyingSender) wait() {
-	ss.waiter.Wait()
-	ss.waiter = nil
+func (ss *notifyingSender) reset(notify chan struct{}) {
+	ss.notify = notify
 }
 
 func (ss *notifyingSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
@@ -67,9 +61,12 @@ func (ss *notifyingSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*
 	if br != nil && br.Error != nil {
 		panic(roachpb.ErrorUnexpectedlySet(ss.wrapped, br))
 	}
-	if ss.waiter != nil {
-		ss.waiter.Done()
+
+	select {
+	case ss.notify <- struct{}{}:
+	default:
 	}
+
 	return br, pErr
 }
 
@@ -164,24 +161,30 @@ func TestClientRetryNonTxn(t *testing.T) {
 			if pErr := txn.Put(key, "txn-value"); pErr != nil {
 				return pErr
 			}
-			// The wait group lets us pause txn until after the non-txn method has run once.
-			wg := sync.WaitGroup{}
 			// On the first true, send the non-txn put or get.
 			if count == 1 {
 				// We use a "notifying" sender here, which allows us to know exactly when the
 				// call has been processed; otherwise, we'd be dependent on timing.
-				sender.reset(&wg)
+				// The channel lets us pause txn until after the non-txn method has run once.
+				// Use a channel length of size 1 to guarantee a notification through a
+				// non-blocking send.
+				notify := make(chan struct{}, 1)
+				sender.reset(notify)
 				// We must try the non-txn put or get in a goroutine because
 				// it might have to retry and will only succeed immediately in
 				// the event we can push.
 				go func() {
 					var pErr *roachpb.Error
-					for i := 0; ; i++ {
+					for {
 						if _, ok := test.args.(*roachpb.GetRequest); ok {
 							_, pErr = db.Get(key)
 						} else {
 							pErr = db.Put(key, "value")
 						}
+						// The above Get/Put() calls Send() which releases
+						// notify below; the txn proceeds to succeed.
+						// The above Get/Put() is repeated until no WriteIntentError
+						// is seen.
 						if _, ok := pErr.GetDetail().(*roachpb.WriteIntentError); !ok {
 							break
 						}
@@ -191,7 +194,7 @@ func TestClientRetryNonTxn(t *testing.T) {
 						t.Fatalf("%d: expected success on non-txn call to %s; got %s", i, test.args.Method(), pErr)
 					}
 				}()
-				sender.wait()
+				<-notify
 			}
 			return nil
 		})

